File: android-utils/anti-android/anti-src/src/utils/permission_utils.cc
//
// Created by amlloc on 18-7-13.
//

//public static boolean checkPermissions(String... perms){
//        Context ctx = ContextFinder.getApplication();
//
//        for(String perm : perms){
//            if(ctx.checkPermission(perm, Process.myPid(), Process.myUid()) != PackageManager.PERMISSION_GRANTED){
//                return false;
//            }
//        }
//
//        return true;
//    }
#include <jni.h>
#include <string>
#include "jni_comm_utils.h"
#include "jni_utils.h"

using namespace std;
bool checkPermissions(const char*  perm){

    DXAnti_JNIEnvGuard guard;
    JNIEnv *env = anti_get_jni_env();
    jobject jcontext = anti_jni_getGlobalContext(env);

    //android.os.Process
    //android.content.Context
    //checkPermission(String var1, int var2, int var3);
    jclass jProcess = anti_jni_findClassByName(env, "android/os/Process");

    jint myPid = anti_jni_callStaticMethodReturnInt(env, jProcess, "myPid", "()I");
    jint myUid = anti_jni_callStaticMethodReturnInt(env, jProcess, "myUid", "()I");
    jstring jperm = env->NewStringUTF(perm);
    jint jretInt = anti_jni_callIntMethodByName(env, jcontext, "checkPermission",
                                                         "(Ljava/lang/String;II)I", jperm, myPid,
                                                         myUid);
    bool isPermitted = true;
    if (jretInt != 0){
        //anti_err_log("permission needed: %s", perm);
        isPermitted = false;
    }

    anti_jni_delete_ref(env, jProcess);
    anti_jni_delete_ref(env, jperm);
    return isPermitted;
}

File: android-utils/risk/as_risk/dx-risk/src/main/java/com/dx/mobile/risk/utils/PermissionUtils.java
package com.dx.mobile.risk.utils;

import android.Manifest;
import android.content.Context;
import android.content.pm.PackageManager;
import android.os.Process;

/**
 * Created by boyliang on 2017/6/11.
 */

public final class PermissionUtils {
//    public static String READ_PHONE_STATE = "android.permission.READ_PHONE_STATE";
//    public static String WRITE_EXTERNAL_STORAGE = "android.permission.WRITE_EXTERNAL_STORAGE";
//    public static String CAMERA = "android.permission.CAMERA";
//    public static String INTERNET = "android.permission.INTERNET";
//    public static String ACCESS_WIFI_STATE = "android.permission.ACCESS_WIFI_STATE";
//    public static String BLUETOOTH = "android.permission.BLUETOOTH";

    public static boolean checkPermissions(String ... perms){
        try{
            Context ctx = ContextFinder.getApplication();
            for(String perm : perms){
                if(ctx.checkPermission(perm, Process.myPid(), Process.myUid()) != PackageManager.PERMISSION_GRANTED){
                    return false;
                }
            }
            return true;
        }catch(Throwable t){
        }

        return false;
        

    }

    //检查是否同时具有读写删文件的权限
    public static boolean checkWRPermission(){
        return checkPermissions(Manifest.permission.WRITE_EXTERNAL_STORAGE,
                Manifest.permission.READ_EXTERNAL_STORAGE);

    }

    public static boolean checkLocation() {
        return checkPermissions(Manifest.permission.ACCESS_FINE_LOCATION,
                Manifest.permission.ACCESS_COARSE_LOCATION);
    }

    public static boolean checkCamerPermission(){
        return checkPermissions(Manifest.permission.CAMERA);
    }
}
